packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -167,133 +244,78 @@ export default {
+        resultIds = entity?.[targetField]?.map((result: any) => result.id);
+      } else {
+        resultIds = entity?.[targetField]?.id ? [entity[targetField].id] : [];
       }
     }
 
-    const targetedModel = strapi.getModel(attribute.target);
-
-    const modelConfig = isComponent
-      ? await getService('components').findConfiguration(modelSchema)
-      : await getService('content-types').findConfiguration(modelSchema);
-
-    const mainField = flow(
-      prop(`metadatas.${targetField}.edit.mainField`),
-      (mainField) => mainField || 'id',
-      (mainField) => sanitizeMainField(targetedModel, mainField, userAbility)
-    )(modelConfig);
-
-    const fieldsToSelect = uniq(['id', mainField, PUBLISHED_AT_ATTRIBUTE]);
-
-    const queryParams = {
-      fields: fieldsToSelect,
-    };
-
-    const permissionChecker = getService('permission-checker').create({
-      userAbility,
-      model: targetedModel.uid,
+    const fields: Array<string> = locale ? [...fieldsToSelect, 'locale'] : fieldsToSelect;
+    const sort = fields
+      .filter((field: any) => !['id', 'locale', 'publishedAt'].includes(field))
+      .map((field: any) => `${field}:ASC`);
+
+    const page = await strapi.entityService.findPage(targetUid as Common.UID.ContentType, {
+      fields,
+      filters: {
+        id: { $in: resultIds },
+      },
+      sort,
+      page: ctx.request.query.page,
+      pageSize: ctx.request.query.pageSize,
     });
-    const permissionQuery = await permissionChecker.sanitizedQuery.read(queryParams);
-
-    if (isAnyToMany(attribute)) {
-      const res = await strapi.entityService.loadPages(
-        model,
-        { id },
-        targetField,
-        {
-          fields: ['id'],
-          ordering: 'desc',
-        } as any,
-        {
-          page: ctx.request.query.page,
-          pageSize: ctx.request.query.pageSize,
-        }
-      );
-      const ids = res.results.map((item: any) => item.id);
-
-      addFiltersClause(permissionQuery, { id: { $in: ids } });
-
-      const sanitizedRes = await strapi.entityService.loadPages(
-        model,
-        { id },
-        targetField,
-        {
-          ...permissionQuery,
-          ordering: 'desc',
-        } as any,
-        {
-          page: ctx.request.query.page,
-          pageSize: ctx.request.query.pageSize,
-        }
-      );
 
-      res.results = uniqBy('id', concat(sanitizedRes.results, res.results));
-
-      ctx.body = res;
-    } else {
-      const [resWithOnlyId, res] = await Promise.all([
-        strapi.entityService.load(model, { id }, targetField, {
-          fields: ['id'],
-        }),
-        strapi.entityService.load(model, { id }, targetField, {
-          ...permissionQuery,
-        }),
-      ]);
-      const result = res || resWithOnlyId;
-      ctx.body = {
-        data: result,
-      };
-    }
+    ctx.body = {
+      ...page,
+      results: Array.isArray(page.results) ? mapResults(page.results) : [],
```

Comment:
I believe the response format for xtoOne might be different than in v4, looks fine by me, but it is frontend aware of the change?

Reply:
@joshuaellis is there a problem with changing the response for xToOne relations? This way the response will always be an array in `res.results`

Reply:
we convert the object format to an array anyway, so if you do it, thats fine, just dont forget to update the contracts 🤓